source/index.erb

```diff
@@ -10,7 +10,25 @@ title: Energy Globe Award
 
   <section ng-controller="AwardsController">
 
-    <div class="award" ng-repeat="award in awards">
+    <section class="filters">
+      <h2>Kategorien</h2>
+      <ul class="category-filter" ng-repeat="category in categories">
+        <li>
+          <input type="checkbox" id="category_checkbox_{{category}}" ng-model="categoryIsShown[category]">
+          <label for="category_checkbox_{{category}}">{{category}}</label>
+        </li>
+      </ul>
+
+      <h2>Jahre</h2>
+      <ul class="year-filter" ng-repeat="year in years">
+        <li>
+          <input type="checkbox" id="year_checkbox_{{year}}" ng-model="yearIsShown[year]">
+          <label for="year_checkbox_{{year}}">{{year}}</label>
+        </li>
+      <ul>
+    </section>
+
+    <div class="award" ng-repeat="award in awards | categoriesFilter:categoryIsShown | yearsFilter: yearIsShown">
```

Comment:
Exactly, I would use angular just for rerendering the filtered list when something changes. The logic to generate the filtered list (our pipeline) should not know anything about angular or rendering. Just imagine this code should work the same way on the server ;-)